packages/patternfly-4/react-core/src/components/Switch/Switch.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Switch/switch.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** id for the label. */
+  id: PropTypes.string.isRequired,
+  /** Additional classes added to the Switch */
+  className: PropTypes.string,
+  /** Text value for the label on the left */
+  labelLeft: PropTypes.string,
+  /** Text value for the label on the right */
+  labelRight: PropTypes.string,
+  /** Flag to show if the Switch is checked. */
+  isChecked: PropTypes.bool,
+  /** Flag to show if the Switch is disabled. */
+  isDisabled: PropTypes.bool,
+  /** A callback for when the Switch selection changes. (isChecked, event) => {} */
+  onChange: PropTypes.func,
+  /** Adds accessible text to the input. */
+  'aria-label': PropTypes.string
+};
+
+const defaultProps = {
+  className: '',
+  labelLeft: '',
+  labelRight: '',
+  isChecked: true,
+  isDisabled: false,
+  'aria-label': ''
+};
+
+class Switch extends React.Component {
+  handleChange = event => {
+    this.props.onChange(event.currentTarget.checked, event);
+  };
+
+  render() {
+    const { id, className, labelLeft, labelRight, isChecked, isDisabled, onChange, ...props } = this.props;
+    return (
+      <label className={css(styles.switch, className)} htmlFor={id}>
+        <input
+          {...props}
+          id={id}
+          className={css(styles.switchInput)}
+          type="checkbox"
+          onChange={onChange ? this.handleChange : null}
```

Comment:
Typically, event should be first param sent back. But looking through some other components it appears that we are not consistent with that.
Also could probably remove the extra function and change to:
`onChange={(event) => onChange(event.currentTarget.checked, event)}`